cmd/clusterctl/examples/openstack/machines.yaml.template

```diff
@@ -1,10 +1,11 @@
 items:
 - apiVersion: "cluster.k8s.io/v1alpha1"
-  kind: Machine
+  kind: MachineList
```

Comment:
@jichenjc Not each element of items is `kind: MachineList`. Each Machine is still `kind: Machine`, but the entire file `kind: MachineList`. See https://github.com/kubernetes-sigs/cluster-api/blob/77224b1d1add00376769b3eeb70a4995e8b22577/pkg/util/util.go#L244

`MachineList` is defined here: https://github.com/kubernetes-sigs/cluster-api/blob/a918956a7660bc1a3058d7a190cc91e09be8612d/pkg/apis/cluster/v1alpha1/machine_types.go#L242


Example:
```
apiVersion: "cluster.k8s.io/v1alpha1"
kind: MachineList
items:
- apiVersion: "cluster.k8s.io/v1alpha1"
  kind: Machine
  metadata:
    generateName: openstack-master-
    labels:
      set: master
      cluster.k8s.io/cluster-name: test1
```

Reply:
ok, I didn't pay attention to this, thanks for the guide and I will read carefully before submit next PR

Reply:
ok, read the code and seems current template should be fine.. this need some test but should be easier to get 0.1.0 onboard to do test..